### PR TITLE
DEV-269: First Class Sitting API

### PR DESCRIPTION
**Pull request by luiscuenca**
https://highfidelity.atlassian.net/browse/DEV-269

A new API has been exposed:

MyAvatar.beginSit 
MyAvatar.endSit

This will simplify the current sitting script.

Since the create app is broken I've created a script that rez 4 seats with the proper scripts attached to them. [The script](https://hifi-content.s3.amazonaws.com/luis/test_scripts/Sitting/sitClientV2.js) that is  attached to the seat entities is the modified version of the sit script V2 that calls `MyAvatar.beginSit` and `MyAvatar.endSit`. I've also commented the code that is not necessary anymore with the new API.

## TEST PLAN
- [This script](https://hifi-content.s3.amazonaws.com/luis/test_scripts/Sitting/createSits.js) will create 4 seats (wireframe boxes) in front of you. 
- Seat by clicking on the icon and stand up with the space key.
- Change seats while seated. This should not make you stand up.
- The initial rotation of the avatar when sits should be the seat's rotation.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20518/HighFidelity-Beta-PR15974-015d195.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20518/HighFidelity-Quest-Beta-PR15974-015d195.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20518/HighFidelity-Beta-PR15974-015d195.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20518/HighFidelity-Beta-Interface-PR15974-015d195.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20534/HighFidelity-Beta-PR15974-8900c4f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20534/HighFidelity-Quest-Beta-PR15974-8900c4f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20534/HighFidelity-Beta-PR15974-8900c4f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20534/HighFidelity-Beta-Interface-PR15974-8900c4f.dmg

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -6207,3 +6207,48 @@ void MyAvatar::sendPacket(const QUuid& entityID) const {
...
+        if (enabled) {
+            enableDriveKey(key);
+        } else {
+            disableDriveKey(key);
+        }
+    }
+}
+
+void MyAvatar::beginSit(const glm::vec3& position, const glm::quat& rotation) {
+    _characterController.setSeated(true);
```
Comment (hyperlogic) [inline interface/src/avatar/MyAvatar.cpp]: So, In the previous sit script many of these items had setTimeouts() between them, ensuring that the effects of these occurred methods took effect on the main thread on one frame, and others on the next.  I'm concerned that we will introduce some new race conditions here.  I suppose we will run into them during testing.

**Comment by hyperlogic** [comment]
Mysterious build failure on windows.
```
13:42:30 Done Building Project "C:\jenkins\workspace\pr-build\label\windows\build\libraries\entities-renderer\entities-renderer.vcxproj" (default targets) -- FAILED.
```
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20541/HighFidelity-Beta-PR15974-8900c4f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20541/HighFidelity-Quest-Beta-PR15974-8900c4f.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20541/HighFidelity-Beta-PR15974-8900c4f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20541/HighFidelity-Beta-Interface-PR15974-8900c4f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20541/HighFidelity-Beta-PR15974-8900c4f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20541/HighFidelity-Beta-Interface-PR15974-8900c4f.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20592/HighFidelity-Beta-PR15974-c1b818c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20592/HighFidelity-Quest-Beta-PR15974-c1b818c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20592/HighFidelity-Beta-PR15974-c1b818c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20592/HighFidelity-Beta-Interface-PR15974-c1b818c.exe

**Comment by luiscuenca** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20596/HighFidelity-Beta-PR15974-c1b818c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20596/HighFidelity-Quest-Beta-PR15974-c1b818c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20596/HighFidelity-Beta-PR15974-c1b818c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20596/HighFidelity-Beta-Interface-PR15974-c1b818c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20596/HighFidelity-Beta-PR15974-c1b818c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20596/HighFidelity-Beta-Interface-PR15974-c1b818c.dmg

**Pull request merged**